binderhub/app.py

```diff
@@ -612,6 +613,41 @@ def _default_build_token_secret(self):
         """
     )
 
+    federation_id = Unicode(
```

Comment:
it feels a bit weird that we are now encoding the notion of a "federation" in BinderHub itself - do we imagine any other usecase than mybinder.org where a user would want this? Or are we OK with designing BinderHub features that are just for mybinder.org?

Reply:
This is the main difference between the two - this PR requires BinderHubs to 'pull' quotas and know about its peers, whereas #1460 uses a 'push' model, which works if we tell all the instances to push their quota consumption to the same place.

I can also adjust #1460 to change less - keep the concurrent-session quota counting instead of _also_ switching to rate limiting. But then frankly, we are starting to reimplement a KV store, and should probably deploy etcd/memcache/redis to handle that.